hunter-packages/ogles_gpgpu
# filter updates and unit tests

* update hunter to v0.18.29
* hunterized unit tests (gtest)
   - add core GTest unit tests (use OpenCV for IO and datatypes)
   - add custom OpenCV CMAKE_ARGS to fix opencv os x build error
   - initial per filter coverage w/ simple setup + teardown + readout test (some value based tests added)
* update .gitignore
* astyle --style=java --recursive ./*.cpp ./*.h (closest match to original formatting)
* add astyle indentiation guards for shader string formatting
* fix new shader author/copyright notice
* move harris corner shader to designated shader
* add simple 3x3 hessian class; output(Ixx, Iyy, Ixy, det(H))
* add rgb2hsv and hsv2rgb shaders
* add simple gain shader
* provide two output options in hessian filter
   + det(H(R)) det(H(G)) det(H(B)) 1.0
   + Ixx(R) Ixy(R), Iyy(R), det(H(R))
* add newline to logging code
* load egl{Create,Destroy,ClientWait}SyncKHR sync operators dynamically, w/ fallback to glFinish()
* add missing libs to target_link_libraries; add missing defs to target_comile_definitions
* make nms output the peak value instead of binary value
* allow control of input and output channel for NMS processing
* add delegates for generic per filter "self reflection"
  + virtual void setPreProcessCallback(ProcDelegate &cb);
  + virtual void setPostProcessCallback(ProcDelegate &cb);
  + virtual void setPreRenderCallback(ProcDelegate &cb);
  + virtual void setPostRenderCallback(ProcDelegate &cb);
  + virtual void setPreInitCallback(ProcDelegate &cb);
  + virtual void setPostInitCallback(ProcDelegate &cb);
* add access method for shader program id
* comments, set inpuPixFormat on yuv->rgb shader initialization
* static analyzer: null assertions, order of initialization
* DiffProc (add missing destructor) : required by Xcode for shared_ptr<DiffProc> use in some cases
* FirProc: add alpha  (scale) and beta (offset) shader uniforms
* remove some custom/experimental shader source
* add missing strength/scale uniform in flow shader
* clear filters from procPasses in destructor
